File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {

    const maHash = {}

    for (const key of nums)
    {
        if (!maHash[key])
            maHash[key] = 0
        else (maHash[key]) 
            maHash[key]++
    }

    const sortHash = []

    for (const hash in maHash)
        sortHash.push([hash, maHash[hash]])

    sortHash.sort((a,b) => b[1] - a[1]);


    const res = []

    for (let i = 0; i < k ; i++)
    {
        res.push(parseInt(sortHash[i][0]))
    }
    return res;

};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int, int> myHash = new Dictionary<int, int>();

        foreach(var n in nums)
        {
            if (!myHash.ContainsKey(n))
                myHash[n] = 1;
            else
                myHash[n]++;
        }

        List<KeyValuePair<int, int>> sortedList = new List<KeyValuePair<int, int>>(myHash);
        sortedList.Sort((x, y) => y.Value.CompareTo(x.Value));


        int[] res = new int[k];
        for (int i = 0; i < k; i++) {
            res[i] = sortedList[i].Key;
        }

        return res;
    }
}